paperqa/docs.py

```diff
@@ -513,7 +517,10 @@ async def _build_texts_index(self, embedding_model: EmbeddingModel) -> None:
                 strict=True,
             ):
                 t.embedding = t_embedding
-        self.texts_index.add_texts_and_embeddings(texts)
+        if asyncio.iscoroutinefunction(self.texts_index.add_texts_and_embeddings):
+            await self.texts_index.add_texts_and_embeddings(texts)
```

Comment:
Ah this is too bad. I agree that `add_texts_and_embeddings` should be `async`, then we wouldn't have this issue.

Do you mind opening a separate/upstream PR that converts `VectorStore.add_texts_and_embeddings` to be `async`?

Reply:
Will open another PR for this.